ansible/roles/fanout/tasks/main.yml

```diff
@@ -6,36 +6,60 @@
 ### specified in this playbook, you would need to come up with your own fanout switch deployment 
 ### playbook
 ################################################################################################
-  # Gather minigraph facts
+# change the root fanout port vlans
+- set_fact:
+    dut: "{{ inventory_hostname }}"
+    connect_leaf: true
+    leaf: "{{ ansible_host }}"
+
+- include: rootfanout_connect.yml
+
+# Deploy leaf fanout switch
+- set_fact:
+    ansible_host: "{{ leaf }}"
+
 - name: Gathering lab graph facts about the device
   conn_graph_facts: host={{ inventory_hostname }}
   connection: local
-  tags: always
 
 - name: prepare fanout switch admin login info
   set_fact: ansible_ssh_user={{ fanout_admin_user }} ansible_ssh_pass={{ fanout_admin_password }}
 
 - block:
+  - name: create persistent shell login
+    shell: touch /mnt/flash/rc.eos
```

Comment:
Why do we need a persistent shell prompt?

Reply:
Well, it depends. for new switch, if manually copied this file over, then no need for this. Otherwise, if just manually add a new user to run deployment, after switch reload, the shell login is gone.